provider/resources.go

```diff
@@ -323,6 +323,12 @@ func lowercaseAutoName() *tfbridge.SchemaInfo {
 	})
 }
 
+func lowercaseAutoNameFields() map[string]*tfbridge.SchemaInfo {
```

Comment:
for readability purposes, I don't love this abstraction since it really only applies to the `name` field.
Currently, it's immediately obvious at a glance per resource that we're doing something about the `name` field specifically.

Reply:
I don't want to duplicate out the entire `map[string]*tfbridge.SchemaInfo{ "name": value }`. It's super long and a great way to introduce a typo. I refactored `lowercaseAutoNameFields()` to `nameField(lowercaseAutoName())`, which is more explicit.